server/repository/repository_test.go

```diff
@@ -90,15 +90,16 @@ var (
 		},
 	}
 	fakeRepo = appsv1.Repository{
-		Repo:      "https://test",
-		Type:      "test",
-		Name:      "test",
-		Username:  "argo",
-		Insecure:  false,
-		EnableLFS: false,
-		EnableOCI: false,
-		Proxy:     "test",
-		Project:   "argocd",
+		Repo:           "https://test",
+		Type:           "test",
+		Name:           "test",
+		Username:       "argo",
+		Insecure:       false,
+		EnableLFS:      false,
+		EnableOCI:      false,
+		Proxy:          "test",
+		Project:        "argocd",
+		InheritedCreds: true,
```

Comment:
Usually the default boolean value is `false`. What is the reasoning behind the assumptions of `InheritedCreds` be `true`?

Reply:
Yes, the default is `false`. So, in order to test that a value is being propagated by the API, we need to test a non-default value. Or, to point it another way, if the attribute were not propagated then the result from any API calls would use the default (`false`) - which is why this was not caught by the existing tests previously. 

FYI: `fakeRepo` is used in the existing test for the `List` endpoint [here](https://github.com/onematchfox/argo-cd/blob/c8c1f3447882579e52fe1bd8c59a91fdd0c2f5e0/server/repository/repository_test.go#L332). So, this specific change was made to ensure that this field is being propagated via that endpoint.